### PR TITLE
Orleans v7 upgrade

### LINKED ISSUE #36: Upgrade to Orleans 7?
Hello! Is this project still alive? Are there any plans to upgrade to Orleans 7? Or is there an alternative pattern that should now be used?

If it's just a matter of capacity, I could take a first crack at a v7 upgrade PR.

### PR DESCRIPTION
**Pull request by htxryan**
# Description
Upgrading to Orleans v7 (and .NET 7).

Fixes #36 

## Type of Change
* [ ] Bug fix
* [ ] New Feature
* [ ] Documentation
* [ ] Code improvement
* [X] Breaking change - if a public API changes, or any change that ***DOES*** or ***MAY*** require a major revision to the NuGet package version due to [semver](https://semver.org/).
* [ ] Unit tests
* [ ] Code samples
* [ ] Added your repository URL to the readme because you make use of this super cool package! ;)
* [X] Other

## Describe testing that was performed for your change
All existing tests are passing.

## Checklist
* [X] Read the https://github.com/OrleansContrib/Orleans.SyncWork/blob/main/CONTRIBUTING.md
* [X] Ran unit tests and ensured they passed
* [ ] Added unit tests where applicable
* [X] Referenced an issue where applicable
* [X] Ran `dotnet-format` locally

### CODE REVIEW

**Comment by Kritner** [comment]
Sorry about the delay, I'll take a look at this shortly!

File: samples/Orleans.SyncWork.Demo.Services/Grains/PasswordVerifier.cs [hunk 1]
```diff
@@ -25,13 +25,19 @@ protected override async Task<PasswordVerifierResult> PerformWork(PasswordVerifi
         };
     }
 }
+
+[GenerateSerializer]
```
Comment (Kritner) [inline samples/Orleans.SyncWork.Demo.Services/Grains/PasswordVerifier.cs:29]: Can you speak to this change as well as the use of `[Id(0)]` and the like?  Is this a means of forcing the generation/order of serialization without having to specify the Configure application parts from samples/Orleans.SyncWork.Demo.Api/OrleansConfigurationHelper.cs ?

File: src/Orleans.SyncWork/Orleans.SyncWork.csproj [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
-    <LangVersion>10</LangVersion>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (Kritner) [inline src/Orleans.SyncWork/Orleans.SyncWork.csproj:4]: is there a reason that we *have* to bring in .net7 dependencies and can't rely on .net standard?  

Not knowing a *whole* lot about how the packages work I'm a bit out of my element here.

Wonder if we can just do different directives for producing packages for both.

On the other hand, not sure if it's that big of a deal, could just say "not doing bug fixes for old versions, only for .net7 and forward"

File: test/Orleans.SyncWork.Tests/SerializationSurrogates/SaltParseExceptionSerializationSurrogate.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using BCrypt.Net;
+
+namespace Orleans.SyncWork.Tests.SerializationSurrogates;
+
+[GenerateSerializer]
+public struct SaltParseExceptionSerializationSurrogate
```
Comment (Kritner) [inline test/Orleans.SyncWork.Tests/SerializationSurrogates/SaltParseExceptionSerializationSurrogate.cs:7]: How does this get used/registered?  Is it just auto magic?

File: samples/Orleans.SyncWork.Demo.Services/Grains/PasswordVerifier.cs (hunk 1 above)
Comment (htxryan) [reply samples/Orleans.SyncWork.Demo.Services/Grains/PasswordVerifier.cs:29]: This is the new default serialization configuration in Orleans v7. You _can_ overwrite the default serializer to use Json.Net, but I figured it was better to just stick with the default. The docs here explain how it works: https://learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans#serialization

File: src/Orleans.SyncWork/Orleans.SyncWork.csproj (hunk 2 above)
Comment (htxryan) [reply src/Orleans.SyncWork/Orleans.SyncWork.csproj:4]: I was unable to build with the upgrade Orleans v7 dependencies without upgrading to net7.0. I'm not 100% sure that there isn't another way around it, but if you look at the Orleans v7 packages they are not listed as being compatible with net standard -- only net7.0.

https://www.nuget.org/packages/Microsoft.Orleans.Server
image

File: test/Orleans.SyncWork.Tests/SerializationSurrogates/SaltParseExceptionSerializationSurrogate.cs (hunk 3 above)
Comment (htxryan) [reply test/Orleans.SyncWork.Tests/SerializationSurrogates/SaltParseExceptionSerializationSurrogate.cs:7]: Yes -- Orleans v7 did away with the explicit "AddApplicationParts()" construct and does assembly scanning automatically. These serializer attributes trigger the Orleans code to generate serializers for these classes. You will see runtime errors if you attempt to pass a class between grains and you've missed the `[GenerateSerializer]` attribute, something along the lines of "could not locate serializer for type".

https://learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans#packaging-changes
https://learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans#serialization

**Comment by Kritner** [comment]
Looks great, thanks @htxryan!  I'll get a release out shortly

**Pull request merged**